xonsh/foreign_shells.py

```diff
@@ -236,7 +245,7 @@ def parse_aliases(s):
     return aliases
 
 
-FUNCS_RE = re.compile('__XONSH_FUNCS_BEG__\n(.*)__XONSH_FUNCS_END__',
+FUNCS_RE = re.compile('__XONSH_FUNCS_BEG__\n(.+)\n__XONSH_FUNCS_END__',
```

Comment:
This is a general bug fix. It ensures that the regular expression is `None` if there are no matches.